sources/prairie-dog/src/main.rs

```diff
@@ -0,0 +1,431 @@
+/*!
+  prairie-dog is a tool to provide kdump support in Bottlerocket. It performs three operations:
+
+  - _digs_ to find the active boot partition and mounts it in /boot
+  - loads the crash kernel from /boot
+  - creates memory dumps when the kernel panics
+*/
+
+#![deny(rust_2018_idioms)]
+
+#[macro_use]
+extern crate log;
+
+use nix;
+use signpost::State;
+use simplelog::{Config as LogConfig, LevelFilter, SimpleLogger, WriteLogger};
+use snafu::{ensure, OptionExt, ResultExt};
+use std::env;
+use std::ffi::OsStr;
+use std::fs::{self, File};
+use std::path::Path;
+use std::process::{self, Command};
+use std::str::FromStr;
+
+// Kdump related binary paths
+const MAKEDUMPFILE_PATH: &str = "/sbin/makedumpfile";
+const KEXEC_PATH: &str = "/sbin/kexec";
+
+// Mount points created prairie-dog
+const LOCAL_BOOT_MOUNT_PATH: &str = "/local/boot";
+const BOOT_MOUNT_PATH: &str = "/boot";
+
+// Mounts configurations
+const BOOT_MOUNT_SELINUX_CONTEXT: &str = "context=system_u:object_r:os_t:s0";
+
+// Files generated by prairie-dog
+const KDUMP_LOGS_PATH: &str = "/var/log/kdump";
+const LOG_FILE: &str = "prairie-dog.log";
+const DMESG_DUMP_FILE: &str = "dmesg.dump";
+const KDUMP_FILE: &str = "kdump.dump";
+
+// Stores how much memory was allocated for the crash kernel
+const KEXEC_CRASH_SIZE: &str = "/sys/kernel/kexec_crash_size";
+// Enables/disables the kexec_load/kexec_file_load syscalls
+const KEXEC_LOAD_DISABLED: &str = "/proc/sys/kernel/kexec_load_disabled";
+
+// Used to pass None to nix::mount::mount
+const NONE: Option<&'static [u8]> = None;
+
+/// Stores user-supplied global arguments
+#[derive(Debug)]
+struct Args {
+    log_level: LevelFilter,
+}
+
+impl Default for Args {
+    fn default() -> Self {
+        Self {
+            log_level: LevelFilter::Info,
+        }
+    }
+}
```

Comment:
You can feel free to use `argh` rather than hand-rolled argument parsing code. Size is a bit of a concern for this binary, given where it runs, but `argh` doesn't cost very much.